mono/mini/local-propagation.c

```diff
@@ -640,11 +640,9 @@ mono_local_cprop (MonoCompile *cfg)
 				}
 
 				/* Constant propagation */
-				/* FIXME: Make is_inst_imm a macro */
-				/* FIXME: Make is_inst_imm take an opcode argument */
 				/* is_inst_imm is only needed for binops */
 				if ((((def->opcode == OP_ICONST) || ((sizeof (gpointer) == 8) && (def->opcode == OP_I8CONST)) || (def->opcode == OP_PCONST)) &&
-					 (((srcindex == 0) && (ins->sreg2 == -1)) || mono_arch_is_inst_imm (def->inst_c0))) || 
```

Comment:
This should be fine since we call `mono_arch_is_inst_imm` below before using immediate opcodes anyway (and several archs just implement it as `return TRUE;` anyway, and do lowering in `mono_arch_lowering_pass` as needed).